assets/global.js

```diff
@@ -679,17 +679,16 @@ class SlideshowComponent extends SliderComponent {
   }
 
   setAutoPlay() {
-    this.sliderAutoplayButton = this.querySelector('.slideshow__autoplay');
+    if (this.querySelector('.slideshow__autoplay') !== null) {
+      this.sliderAutoplayButton.addEventListener('click', this.autoPlayToggle.bind(this));
+      this.addEventListener('mouseover', this.focusInHandling.bind(this));
+      this.addEventListener('mouseleave', this.focusOutHandling.bind(this));
+      this.addEventListener('focusin', this.focusInHandling.bind(this));
+      this.addEventListener('focusout', this.focusOutHandling.bind(this));
+      this.autoplayButtonIsSetToPlay = true;
+    }
```

Comment:
This might be a slightly cleaner option if you prefer
```suggestion
    if (this.querySelector('.slideshow__autoplay') === null) return;
 
    this.sliderAutoplayButton.addEventListener('click', this.autoPlayToggle.bind(this));
    this.addEventListener('mouseover', this.focusInHandling.bind(this));
    this.addEventListener('mouseleave', this.focusOutHandling.bind(this));
    this.addEventListener('focusin', this.focusInHandling.bind(this));
    this.addEventListener('focusout', this.focusOutHandling.bind(this));
    this.autoplayButtonIsSetToPlay = true;
```

Reply:
I could be mistaken,  didn't examine in-context. I just assumed if this.querySelector('.slideshow__autoplay') === null anyway, we wouldn't need to call play() or set a value for this.autoplaySpeed

Reply:
The thing is that I don't use this `.slideshow__autoplay` class here. We apply this class in the slideshow component to pause/play buttons which I don't have here. That's why I added that logic above. So, in our case for announcement slideshow `this.querySelector('.slideshow__autoplay'` will always be `null`.